### PR TITLE
[RNMobile] Scroll header toolbar to end when RTL

### PR DESCRIPTION
**Pull request by hypest**
## Description
Fixes the initial horizontal positioning of the main header toolbar when RTL.

### Related PRs
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/3923

## How has this been tested?
Using the gb-mobile PR

## Screenshots 

https://user-images.githubusercontent.com/1032913/132338896-92d7abe5-11ef-4705-9496-d1369c163a6e.mp4

## Types of changes
* `HeaderToolbar` is executing a `scrollToStart` function when the content size changes, attempting to always make sure the inserter button gets visible. This PR updates the `scrollToStart` to respect the RTL setting.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by SiobhyB — APPROVED** [review]
LGTM 🚢 🚀 

I tested this on an Android emulator and confirmed the changes fix the issue. 🎉 I also created [an installable build for iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/17128#issuecomment-914458213) (as I was having trouble seeing the RTL change on the iOS emulator) and confirmed there are no unwanted side effects on iOS when testing with a physical iPhone XR.

I'm curious why this worked as expected on iOS both with and without this change. Is there a known reason I'm missing or perhaps a quirk with the way `ScrollView` works? Either way, the change looks good to my eyes and I've approved it. :D

**Comment by hypest** [comment]
Thanks for the review @SiobhyB and for preparing an installable build of WPiOS!

> I'm curious why this worked as expected on iOS both with and without this change. Is there a known reason I'm missing or perhaps a quirk with the way ScrollView works? Either way, the change looks good to my eyes and I've approved it. :D

Good point! When I tested using a physical iPhone against Metro it looked indeed that the changes here work, but your note now prompted me to check deeper.
I tried the simulator and by using the "Right to Left Pseudolanguage" from the [Testing Using Pseudolanguage](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/TestingYourInternationalApp/TestingYourInternationalApp.html) guide I see that the fix here doesn't work OK (it scrolls the toolbar to the right when on RTL). I'm preparing an update to only target the fix to Android and will push.

Do you mind if I also update your test WPiOS PR alongside?

**Comment by SiobhyB** [comment]
> I tried the simulator and by using the "Right to Left Pseudolanguage" from the [Testing Using Pseudolanguage](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/TestingYourInternationalApp/TestingYourInternationalApp.html) guide I see that the fix here doesn't work OK (it scrolls the toolbar to the right when on RTL). I'm preparing an update to only target the fix to Android and will push.

Ah, really interesting! Thank you for digging deeper and sharing that. I'll read through and bear in mind for the future.

> Do you mind if I also update your test WPiOS PR alongside?

Not at all, feel welcome to update the PR in any way.

**Comment by hypest** [comment]
👋 @SiobhyB ! Ready for another review! I updated all the branches, included the commit to only apply the fix to Android and have triggered an [installable build for iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/17128#issuecomment-914432197). Thanks!

File: packages/react-native-editor/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ For each user feature we should also add a importance categorization label  to i
 -   [**] [Embed block] Implement WP embed preview component [#34004]
 -   [*] [Embed block] Fix content disappearing on Android when switching light/dark mode [#34207]
 -   [*] Embed block: Add device's locale to preview content [#33858]
+-   [**] Fix main toolbar initial position when RTL [#34617]
```
Comment (SiobhyB) [inline packages/react-native-editor/CHANGELOG.md]:
```suggestion
-   [**] [Android only] Fix main toolbar initial position when RTL [#34617]
```

Perhaps it'd be worth adding an `Android only` indicator here, for clarity?

File: packages/edit-post/src/components/header/header-toolbar/index.native.js [hunk 2]
```diff
@@ -43,7 +43,13 @@ function HeaderToolbar( {
 } ) {
 	const scrollViewRef = useRef( null );
 	const scrollToStart = () => {
-		scrollViewRef.current.scrollTo( { x: 0 } );
+		// scrollview doesn't seem to automatically adjust to RTL on Android so, scroll to end when Android
+		const isAndroid = Platform.OS === 'android';
+		if ( isAndroid && isRTL ) {
```
Comment (SiobhyB) [inline packages/edit-post/src/components/header/header-toolbar/index.native.js:48]: Not feedback, but a question: Is the decision to use an if/else statement vs. the ternary operator part of choosing verbosity for the sake of readability we've discussed before? Do have any rules of thumb you go by when making these type of decisions?

**Review by SiobhyB — APPROVED** [review]
LGTM! I tested again on the Android demo app and with the iOS installable build. All worked smoothly in my testing 🎉 🚀 

I left a question and one suggestion for the CHANGELOG, approving this PR as neither of those are blockers and this looks good!

File: packages/react-native-editor/CHANGELOG.md (hunk 1 above)
Comment (hypest) [reply packages/react-native-editor/CHANGELOG.md]: Aha, good point.

May I suggest the following iteration/wording to make it clear that is issue is Android only, not necessarily the solution:

> Fix Android-only issue of main toolbar initial position being wrong when RTL [#34617]

WDYT?

File: packages/edit-post/src/components/header/header-toolbar/index.native.js (hunk 2 above)
Comment (hypest) [reply packages/edit-post/src/components/header/header-toolbar/index.native.js:48]: Oh, sorry I didn't make it clear in the commit message. The change from ternary to if/else was for fixing the [related lint error](https://github.com/WordPress/gutenberg/runs/3533347878#step:5:76).

File: packages/react-native-editor/CHANGELOG.md (hunk 1 above)
Comment (SiobhyB) [reply packages/react-native-editor/CHANGELOG.md]: > to make it clear that is issue is Android only, not necessarily the solution

@hypest, I agree that's a good point and iteration for better communicating the change with testers!

File: packages/edit-post/src/components/header/header-toolbar/index.native.js (hunk 2 above)
Comment (SiobhyB) [reply packages/edit-post/src/components/header/header-toolbar/index.native.js:48]: Aha, I see, thank you for that clarity!

File: packages/react-native-editor/CHANGELOG.md (hunk 1 above)
Comment (hypest) [reply packages/react-native-editor/CHANGELOG.md]: Updated with https://github.com/WordPress/gutenberg/pull/34617/commits/0a88109d91b26f935400e4668b4781dc9bfb2dd2

**Pull request merged**